### DISCUSSION TITLE
Weird Training Plots and No Audio After 70K Steps - Taco1 GST Blizzard

### DISCUSSION THREAD

**Discussion by a-froghyar** [coqui-ai/TTS#377] [General Q&A]
Hey, I'm getting weird plots and no audio produced in the tensorboard examples after 70k Steps. I'm using a custom Blizzard dataset that I've already trained other models with that produced intelligible speech after 20k steps. The training has also stopped after 70K steps because of the NaN decoder_loss error.  I'm using the https://github.com/coqui-ai/TTS/pull/373 patched `dev` branch with the following config file:
{
    "model": "Tacotron",
    "run_name": "blizzard-gts",
    "run_description": "tacotron GST.",
    "audio": {
        
        "fft_size": 1024, 
        "win_length": 1024, 
        "hop_length": 256, 
        "frame_length_ms": null, 
        "frame_shift_ms": null, 

        
        "sample_rate": 24000, 
        "preemphasis": 0.0, 
        "ref_level_db": 20, 

        
        "do_trim_silence": true, 
        "trim_db": 60, 

        
        "power": 1.5, 
        "griffin_lim_iters": 60, 

        
        "num_mels": 80, 
        "mel_fmin": 95.0, 
        "mel_fmax": 12000.0, 
        "spec_gain": 1,

        
        "signal_norm": true, 
        "min_level_db": -100, 
        "symmetric_norm": true, 
        "max_norm": 4.0, 
        "clip_norm": true, 
        "stats_path": null 
    },

    
    "distributed": {
        "backend": "nccl",
        "url": "tcp:\/\/localhost:54321"
    },

    "reinit_layers": [], 

    
    "batch_size": 128, 
    "eval_batch_size": 16,
    "r": 7, 
    "gradual_training": [
        [0, 7, 64],
        [1, 5, 64],
        [50000, 3, 32],
        [130000, 2, 32],
        [290000, 1, 32]
    ], 
    "mixed_precision": true, 

    
    "loss_masking": false, 
    "decoder_loss_alpha": 0.5, 
    "postnet_loss_alpha": 0.25, 
    "postnet_diff_spec_alpha": 0.25, 
    "decoder_diff_spec_alpha": 0.25, 
    "decoder_ssim_alpha": 0.5, 
    "postnet_ssim_alpha": 0.25, 
    "ga_alpha": 5.0, 
    "stopnet_pos_weight": 15.0, 

    
    "run_eval": true,
    "test_delay_epochs": 10, 
    "test_sentences_file": null, 

    
    "noam_schedule": false, 
    "grad_clip": 1.0, 
    "epochs": 300000, 
    "lr": 0.0001, 
    "wd": 0.000001, 
    "warmup_steps": 4000, 
    "seq_len_norm": false, 

    
    "memory_size": -1, 
    "prenet_type": "original", 
    "prenet_dropout": true, 

    
    "attention_type": "graves", 
    "attention_heads": 4, 
    "attention_norm": "sigmoid", 
    "windowing": false, 
    "use_forward_attn": false, 
    "forward_attn_mask": false, 
    "transition_agent": false, 
    "location_attn": true, 
    "bidirectional_decoder": false, 
    "double_decoder_consistency": false, 
    "ddc_r": 7, 

    
    "stopnet": true, 
    "separate_stopnet": true, 

    
    "print_step": 25, 
    "tb_plot_step": 100, 
    "print_eval": false, 
    "save_step": 5000, 
    "checkpoint": true, 
    "tb_model_param_stats": false, 

    
    "text_cleaner": "phoneme_cleaners",
    "enable_eos_bos_chars": false, 
    "num_loader_workers": 8, 
    "num_val_loader_workers": 8, 
    "batch_group_size": 4, 
    "min_seq_len": 6, 
    "max_seq_len": 153, 
    "compute_input_seq_cache": false, 
    "use_noise_augment": true,

    
    "output_path": "/home/big-boy/Models/Blizzard/",

    
    "phoneme_cache_path": "/home/big-boy/Models/phoneme_cache/", 
    "use_phonemes": true, 
    "phoneme_language": "en-us", 

    
    "use_speaker_embedding": false, 
    "use_gst": true, 
    "use_external_speaker_embedding_file": false, 
    "external_speaker_embedding_file": "../../speakers-vctk-en.json", 
    "gst": { 
        "gst_style_input": null, 
        "gst_embedding_dim": 512,
        "gst_num_heads": 4,
        "gst_style_tokens": 10,
        "gst_use_speaker_embedding": false
    },

    
    "datasets": 
        [{
            "name": "ljspeech",
            "path": "/home/big-boy/Data/blizzard2013/segmented/",
            "meta_file_train": "metadata.csv", 
            "meta_file_val": null
        }]
}

![Screenshot from 2021-03-13 11-05-31](https://user-images.githubusercontent.com/23213052/111026714-130be800-83ec-11eb-96ba-5590e99ff1de.png)

![Screenshot from 2021-03-13 10-49-45](https://user-images.githubusercontent.com/23213052/111032114-05b22600-840b-11eb-9b03-eb436c1dcc1a.png)

**Comment by erogol** [comment]
It is not a bug but mostly a training discussion so I move it .

**Comment by a-froghyar** [comment]
Thanks @erogol, I've just attached the training plots as well, the loss seems to just increase from the beginning?

**Comment by a-froghyar** [comment]
Here are some more figures:

![Screenshot from 2021-03-13 15-41-32](https://user-images.githubusercontent.com/23213052/111033671-a8ba6e00-8412-11eb-92f3-de9e4bd302d6.png)

**Reply by erogol** [reply]
your specs are broken

**Reply by erogol** [reply]
you need to compute scale stats on your dataset to normalize spectrograms or use spec_gain=20 but then you cannot use the released vocoder with your model .

**Reply by erogol** [reply]
```bin/compute_statistics.py``` to compute the stats .

**Reply by a-froghyar** [reply]
Thanks so much!

**Reply by erogol** [reply]
what is your dataset BTW? is it LJSpeech?

**Reply by a-froghyar** [reply]
No, it's Blizzard 2013, not the lo-fi but the higher fidelity, segmented stuff

**Reply by a-froghyar** [reply]
But it's in the same format like LJSpeech, that's why I used LJSpeech in the config

**Reply by erogol** [reply]
"mel_fmin": 95.0,    "mel_fmax": 12000.0,  shoudl also be consistent with the released vocoders. They are not currently. Check the released models if this is your  intention to use them .

**Reply by a-froghyar** [reply]
@erogol AFAIK there's no released vocoder for the blizzard 2013 dataset, so I'd have to train one anyways, no? That means I could use the new spectrograms for the vocoder training later on?

**Reply by erogol** [reply]
yes if this is what you want but you can use the released universal vocoders until you train a new one.

**Reply by a-froghyar** [reply]
Hey  @erogol I've set spec_gain=20 and retrained. I now have audio, however the alignment plots are empty. You can find the tensorboard here:  https://tensorboard.dev/experiment/xQt8xPYNTWWQeyIwgOIlgA/

and some alignment plots:

![Screenshot from 2021-03-16 11-05-21](https://user-images.githubusercontent.com/23213052/111291525-92492800-8647-11eb-8ddd-d26035dcd616.png)

Thanks a lot!

**Comment by erogol** [comment]
I can't tell what is wrong just looking at it. I am not at that level yet :). So just debug things yourself and make sure things are fair and square.

**Reply by a-froghyar** [reply]
Hey @erogol, thanks for your reply. I'm going through the steps, also trying to infer some sentences, however I get the following error: AttributeError: 'GravesAttention' object has no attribute 'init_win_idx'. I used GravesAttention also during draining, maybe the alignments are not getting registered because of this? Error comes from layers/ tacotorn.py --> 478  self.attention.init_win_idx()

**Reply by erogol** [reply]
it might be another bug. Pls create and issue explaining the situation.

**Reply by a-froghyar** [reply]
Thank you, I just did: https://github.com/coqui-ai/TTS/issues/383
